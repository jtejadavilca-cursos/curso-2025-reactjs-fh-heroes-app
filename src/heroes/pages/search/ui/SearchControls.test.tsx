import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { beforeEach, describe, expect, test, vi } from "vitest";
import { SearchControls } from "./SearchControls";

if (typeof window.ResizeObserver === "undefined") {
    class ResizeObserver {
        observe() {}
        unobserve() {}
        disconnect() {}
    }
    window.ResizeObserver = ResizeObserver;
    //Object.defineProperty(window, "ResizeObserver", ResizeObserver);
}

const renderSearchControlWithRouter = (initialEntries: string[] = ["/"]) => {
    return render(
        <MemoryRouter initialEntries={initialEntries}>
            <SearchControls />
        </MemoryRouter>
    );
};

describe("Testing SearchControls.test", () => {
    beforeEach(() => {
        vi.clearAllMocks();
    });

    test("should be defined", () => {
        const { container } = renderSearchControlWithRouter();

        expect(container).toMatchSnapshot();
    });

    test("should be advance filters panel opened", () => {
        const { container } = renderSearchControlWithRouter(["?active-accordion=advance-filters-panel&name=Batman"]);

        // Advance filters
        const titleFilterPanel = screen.getByText("Advanced Filters");
        expect(titleFilterPanel).toBeDefined();
        expect(titleFilterPanel.tagName).toBe("H3");

        const btnClearAll = screen.getByText("Clear All");
        expect(btnClearAll).toBeDefined();
        expect(btnClearAll.tagName).toBe("BUTTON");

        const strengthControlLabel = screen.getByText("Minimum Strength: 0/10");
        expect(strengthControlLabel).toBeDefined();
        expect(strengthControlLabel.tagName).toBe("LABEL");

        // Input search
        const inputSearch = screen.getByPlaceholderText("Search heroes, villains, powers, teams...");
        expect(inputSearch.getAttribute("value")).toBe("Batman");

        expect(container).toMatchSnapshot();
    });

    test("should change params when input is changed and enter is pressed", () => {
        const { container } = renderSearchControlWithRouter(["?name=Batman"]);
        const inputSearch = screen.getByPlaceholderText("Search heroes, villains, powers, teams...");

        expect(inputSearch.getAttribute("value")).toBe("Batman");
        fireEvent.change(inputSearch, { target: { value: "Superman" } });
        fireEvent.keyDown(inputSearch, { key: "Enter" });

        expect(inputSearch.getAttribute("value")).toBe("Superman");
        expect(container).toMatchSnapshot();
    });

    test("should change params strength when slider is changed", () => {
        renderSearchControlWithRouter(["/?active-accordion=advance-filters-panel"]);

        const slider = screen.getByRole("slider");
        expect(slider.getAttribute("aria-valuenow")).toBe("0");

        fireEvent.keyDown(slider, { key: "ArrowRight" });
        fireEvent.keyDown(slider, { key: "ArrowRight" });
        fireEvent.keyDown(slider, { key: "ArrowRight" });
        expect(slider.getAttribute("aria-valuenow")).toBe("3");

        fireEvent.keyDown(slider, { key: "ArrowLeft" });
        expect(slider.getAttribute("aria-valuenow")).toBe("2");
    });

    test("should close accordion when filter button is clicked", () => {
        renderSearchControlWithRouter(["/?active-accordion=advance-filters-panel"]);
        // Advance filters is open
        const titleFilterPanel = screen.getByText("Advanced Filters");
        expect(titleFilterPanel).toBeDefined();
        expect(titleFilterPanel.tagName).toBe("H3");

        const btnClearAll = screen.getByText("Clear All");
        expect(btnClearAll).toBeDefined();
        expect(btnClearAll.tagName).toBe("BUTTON");

        const strengthControlLabel = screen.getByText("Minimum Strength: 0/10");
        expect(strengthControlLabel).toBeDefined();
        expect(strengthControlLabel.tagName).toBe("LABEL");

        // Closing advance filters
        const btnFilters = screen.getByText("Filters");
        fireEvent.click(btnFilters);
        expect(screen.queryByText("Clear All")).toBeNull();
        expect(screen.queryByText("Advanced Filters")).toBeNull();
    });
});
