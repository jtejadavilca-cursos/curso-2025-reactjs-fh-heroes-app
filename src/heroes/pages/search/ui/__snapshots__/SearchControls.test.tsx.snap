// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing SearchControls.test > should be advance filters panel opened 1`] = `
<div>
  <div
    class="flex flex-col lg:flex-row gap-4 mb-8"
  >
    <div
      class="relative flex-1"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-search absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-400 h-5 w-5"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m21 21-4.34-4.34"
        />
        <circle
          cx="11"
          cy="11"
          r="8"
        />
      </svg>
      <input
        class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex w-full min-w-0 rounded-md border bg-transparent px-3 py-1 shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive pl-12 h-12 text-lg"
        data-slot="input"
        placeholder="Search heroes, villains, powers, teams..."
        value="Batman"
      />
    </div>
    <div
      class="flex gap-2"
    >
      <button
        class="cursor-pointer inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 px-4 py-2 has-[>svg]:px-3 h-12"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-funnel h-4 w-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 20a1 1 0 0 0 .553.895l2 1A1 1 0 0 0 14 21v-7a2 2 0 0 1 .517-1.341L21.74 4.67A1 1 0 0 0 21 3H3a1 1 0 0 0-.742 1.67l7.225 7.989A2 2 0 0 1 10 14z"
          />
        </svg>
        Filters
      </button>
      <button
        class="cursor-pointer inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 px-4 py-2 has-[>svg]:px-3 h-12"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-arrow-up-narrow-wide h-4 w-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m3 8 4-4 4 4"
          />
          <path
            d="M7 4v16"
          />
          <path
            d="M11 12h4"
          />
          <path
            d="M11 16h7"
          />
          <path
            d="M11 20h10"
          />
        </svg>
        Sort by Name
      </button>
      <button
        class="cursor-pointer inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 px-4 py-2 has-[>svg]:px-3 h-12"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-grid3x3 lucide-grid-3x3 h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="18"
            rx="2"
            width="18"
            x="3"
            y="3"
          />
          <path
            d="M3 9h18"
          />
          <path
            d="M3 15h18"
          />
          <path
            d="M9 3v18"
          />
          <path
            d="M15 3v18"
          />
        </svg>
      </button>
      <button
        class="cursor-pointer inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 px-4 py-2 has-[>svg]:px-3 h-12"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-plus h-4 w-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="M12 5v14"
          />
        </svg>
        Add Character
      </button>
    </div>
  </div>
  <div
    data-orientation="vertical"
    data-slot="accordion"
  >
    <div
      class="border-b last:border-b-0"
      data-orientation="vertical"
      data-slot="accordion-item"
      data-state="open"
    >
      <div
        aria-labelledby="radix-«r2»"
        class="data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down overflow-hidden text-sm"
        data-orientation="vertical"
        data-slot="accordion-content"
        data-state="open"
        id="radix-«r3»"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width); transition-duration: 0s; animation-name: none;"
      >
        <div
          class="pt-0 pb-4"
        >
          <div
            class="bg-white rounded-lg p-6 mb-8 shadow-sm border"
          >
            <div
              class="flex justify-between items-center mb-4"
            >
              <h3
                class="text-lg font-semibold"
              >
                Advanced Filters
              </h3>
              <button
                class="cursor-pointer inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 h-9 px-4 py-2 has-[>svg]:px-3"
                data-slot="button"
              >
                Clear All
              </button>
            </div>
            <div
              class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-4"
            >
              <div
                class="space-y-2"
              >
                <label
                  class="text-sm font-medium"
                >
                  Team
                </label>
                <div
                  class="h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm"
                >
                  All teams
                </div>
              </div>
              <div
                class="space-y-2"
              >
                <label
                  class="text-sm font-medium"
                >
                  Category
                </label>
                <div
                  class="h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm"
                >
                  All categories
                </div>
              </div>
              <div
                class="space-y-2"
              >
                <label
                  class="text-sm font-medium"
                >
                  Universe
                </label>
                <div
                  class="h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm"
                >
                  All universes
                </div>
              </div>
              <div
                class="space-y-2"
              >
                <label
                  class="text-sm font-medium"
                >
                  Status
                </label>
                <div
                  class="h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm"
                >
                  All statuses
                </div>
              </div>
            </div>
            <div
              class="mt-4"
            >
              <label
                class="text-sm font-medium"
              >
                Minimum Strength: 
                0
                /10
              </label>
              <div
                class="relative flex w-full touch-none select-none items-center mt-2 py-5"
              >
                <span
                  aria-disabled="false"
                  class="relative flex w-full touch-none items-center select-none data-[disabled]:opacity-50 data-[orientation=vertical]:h-full data-[orientation=vertical]:min-h-44 data-[orientation=vertical]:w-auto data-[orientation=vertical]:flex-col"
                  data-orientation="horizontal"
                  data-slot="slider"
                  dir="ltr"
                  style="--radix-slider-thumb-transform: translateX(-50%);"
                >
                  <span
                    class="bg-muted relative grow overflow-hidden rounded-full data-[orientation=horizontal]:h-1.5 data-[orientation=horizontal]:w-full data-[orientation=vertical]:h-full data-[orientation=vertical]:w-1.5"
                    data-orientation="horizontal"
                    data-slot="slider-track"
                  >
                    <span
                      class="bg-primary absolute data-[orientation=horizontal]:h-full data-[orientation=vertical]:w-full"
                      data-orientation="horizontal"
                      data-slot="slider-range"
                      style="left: 0%; right: 100%;"
                    />
                  </span>
                  <span
                    style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
                  >
                    <span
                      aria-orientation="horizontal"
                      aria-valuemax="10"
                      aria-valuemin="0"
                      aria-valuenow="0"
                      class="border-primary bg-background ring-ring/50 block size-4 shrink-0 rounded-full border shadow-sm transition-[color,box-shadow] hover:ring-4 focus-visible:ring-4 focus-visible:outline-hidden disabled:pointer-events-none disabled:opacity-50"
                      data-orientation="horizontal"
                      data-radix-collection-item=""
                      data-slot="slider-thumb"
                      role="slider"
                      style=""
                      tabindex="0"
                    />
                  </span>
                  <span
                    style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
                  >
                    <span
                      aria-orientation="horizontal"
                      aria-valuemax="10"
                      aria-valuemin="0"
                      class="border-primary bg-background ring-ring/50 block size-4 shrink-0 rounded-full border shadow-sm transition-[color,box-shadow] hover:ring-4 focus-visible:ring-4 focus-visible:outline-hidden disabled:pointer-events-none disabled:opacity-50"
                      data-orientation="horizontal"
                      data-radix-collection-item=""
                      data-slot="slider-thumb"
                      role="slider"
                      style="display: none;"
                      tabindex="0"
                    />
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing SearchControls.test > should be defined 1`] = `
<div>
  <div
    class="flex flex-col lg:flex-row gap-4 mb-8"
  >
    <div
      class="relative flex-1"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-search absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-400 h-5 w-5"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m21 21-4.34-4.34"
        />
        <circle
          cx="11"
          cy="11"
          r="8"
        />
      </svg>
      <input
        class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex w-full min-w-0 rounded-md border bg-transparent px-3 py-1 shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive pl-12 h-12 text-lg"
        data-slot="input"
        placeholder="Search heroes, villains, powers, teams..."
        value=""
      />
    </div>
    <div
      class="flex gap-2"
    >
      <button
        class="cursor-pointer inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 px-4 py-2 has-[>svg]:px-3 h-12"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-funnel h-4 w-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 20a1 1 0 0 0 .553.895l2 1A1 1 0 0 0 14 21v-7a2 2 0 0 1 .517-1.341L21.74 4.67A1 1 0 0 0 21 3H3a1 1 0 0 0-.742 1.67l7.225 7.989A2 2 0 0 1 10 14z"
          />
        </svg>
        Filters
      </button>
      <button
        class="cursor-pointer inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 px-4 py-2 has-[>svg]:px-3 h-12"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-arrow-up-narrow-wide h-4 w-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m3 8 4-4 4 4"
          />
          <path
            d="M7 4v16"
          />
          <path
            d="M11 12h4"
          />
          <path
            d="M11 16h7"
          />
          <path
            d="M11 20h10"
          />
        </svg>
        Sort by Name
      </button>
      <button
        class="cursor-pointer inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 px-4 py-2 has-[>svg]:px-3 h-12"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-grid3x3 lucide-grid-3x3 h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="18"
            rx="2"
            width="18"
            x="3"
            y="3"
          />
          <path
            d="M3 9h18"
          />
          <path
            d="M3 15h18"
          />
          <path
            d="M9 3v18"
          />
          <path
            d="M15 3v18"
          />
        </svg>
      </button>
      <button
        class="cursor-pointer inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 px-4 py-2 has-[>svg]:px-3 h-12"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-plus h-4 w-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="M12 5v14"
          />
        </svg>
        Add Character
      </button>
    </div>
  </div>
  <div
    data-orientation="vertical"
    data-slot="accordion"
  >
    <div
      class="border-b last:border-b-0"
      data-orientation="vertical"
      data-slot="accordion-item"
      data-state="closed"
    >
      <div
        aria-labelledby="radix-«r0»"
        class="data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down overflow-hidden text-sm"
        data-orientation="vertical"
        data-slot="accordion-content"
        data-state="closed"
        hidden=""
        id="radix-«r1»"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</div>
`;

exports[`Testing SearchControls.test > should change params when input is changed and enter is pressed 1`] = `
<div>
  <div
    class="flex flex-col lg:flex-row gap-4 mb-8"
  >
    <div
      class="relative flex-1"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-search absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-400 h-5 w-5"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m21 21-4.34-4.34"
        />
        <circle
          cx="11"
          cy="11"
          r="8"
        />
      </svg>
      <input
        class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex w-full min-w-0 rounded-md border bg-transparent px-3 py-1 shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive pl-12 h-12 text-lg"
        data-slot="input"
        placeholder="Search heroes, villains, powers, teams..."
        value="Superman"
      />
    </div>
    <div
      class="flex gap-2"
    >
      <button
        class="cursor-pointer inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 px-4 py-2 has-[>svg]:px-3 h-12"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-funnel h-4 w-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 20a1 1 0 0 0 .553.895l2 1A1 1 0 0 0 14 21v-7a2 2 0 0 1 .517-1.341L21.74 4.67A1 1 0 0 0 21 3H3a1 1 0 0 0-.742 1.67l7.225 7.989A2 2 0 0 1 10 14z"
          />
        </svg>
        Filters
      </button>
      <button
        class="cursor-pointer inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 px-4 py-2 has-[>svg]:px-3 h-12"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-arrow-up-narrow-wide h-4 w-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m3 8 4-4 4 4"
          />
          <path
            d="M7 4v16"
          />
          <path
            d="M11 12h4"
          />
          <path
            d="M11 16h7"
          />
          <path
            d="M11 20h10"
          />
        </svg>
        Sort by Name
      </button>
      <button
        class="cursor-pointer inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 px-4 py-2 has-[>svg]:px-3 h-12"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-grid3x3 lucide-grid-3x3 h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="18"
            rx="2"
            width="18"
            x="3"
            y="3"
          />
          <path
            d="M3 9h18"
          />
          <path
            d="M3 15h18"
          />
          <path
            d="M9 3v18"
          />
          <path
            d="M15 3v18"
          />
        </svg>
      </button>
      <button
        class="cursor-pointer inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 px-4 py-2 has-[>svg]:px-3 h-12"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-plus h-4 w-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="M12 5v14"
          />
        </svg>
        Add Character
      </button>
    </div>
  </div>
  <div
    data-orientation="vertical"
    data-slot="accordion"
  >
    <div
      class="border-b last:border-b-0"
      data-orientation="vertical"
      data-slot="accordion-item"
      data-state="closed"
    >
      <div
        aria-labelledby="radix-«r4»"
        class="data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down overflow-hidden text-sm"
        data-orientation="vertical"
        data-slot="accordion-content"
        data-state="closed"
        hidden=""
        id="radix-«r5»"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</div>
`;
